error: this trait bound is already specified in the where clause
  --> $DIR/trait_duplication_in_bounds_unfixable.rs:7:15
   |
LL | fn bad_foo<T: Clone + Default, Z: Copy>(arg0: T, arg1: Z)
   |               ^^^^^
   |
   = help: consider removing this trait bound
note: the lint level is defined here
  --> $DIR/trait_duplication_in_bounds_unfixable.rs:1:9
   |
LL | #![deny(clippy::trait_duplication_in_bounds)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: this trait bound is already specified in the where clause
  --> $DIR/trait_duplication_in_bounds_unfixable.rs:7:23
   |
LL | fn bad_foo<T: Clone + Default, Z: Copy>(arg0: T, arg1: Z)
   |                       ^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound is already specified in trait declaration
  --> $DIR/trait_duplication_in_bounds_unfixable.rs:36:15
   |
LL |         Self: Default;
   |               ^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound is already specified in trait declaration
  --> $DIR/trait_duplication_in_bounds_unfixable.rs:50:15
   |
LL |         Self: Default + Clone;
   |               ^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound is already specified in trait declaration
  --> $DIR/trait_duplication_in_bounds_unfixable.rs:56:15
   |
LL |         Self: Default + Clone;
   |               ^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound is already specified in trait declaration
  --> $DIR/trait_duplication_in_bounds_unfixable.rs:56:25
   |
LL |         Self: Default + Clone;
   |                         ^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound is already specified in trait declaration
  --> $DIR/trait_duplication_in_bounds_unfixable.rs:59:15
   |
LL |         Self: Default;
   |               ^^^^^^^
   |
   = help: consider removing this trait bound

error: this trait bound is already specified in trait declaration
  --> $DIR/trait_duplication_in_bounds_unfixable.rs:94:15
   |
LL |         Self: Iterator<Item = Foo>,
   |               ^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider removing this trait bound

error: aborting due to 8 previous errors

